scrapy/core/downloader/handlers/s3.py

```diff
@@ -31,22 +31,34 @@ def _mexe(self, http_request, *args, **kwargs):
 
 class S3DownloadHandler(object):
 
-    def __init__(self, settings, aws_access_key_id=None, aws_secret_access_key=None, \
+    _conn = None
+
+    def __init__(self, settings, aws_access_key_id=None, aws_secret_access_key=None,
             httpdownloadhandler=HTTPDownloadHandler):
         if 'boto' not in optional_features:
             raise NotConfigured("missing boto library")
 
-        if not aws_access_key_id:
-            aws_access_key_id = settings['AWS_ACCESS_KEY_ID']
-        if not aws_secret_access_key:
-            aws_secret_access_key = settings['AWS_SECRET_ACCESS_KEY']
+        self.aws_access_key_id = aws_access_key_id or settings['AWS_ACCESS_KEY_ID']
+        self.aws_secret_access_key = aws_secret_access_key or settings['AWS_SECRET_ACCESS_KEY']
 
-        try:
-            self.conn = _S3Connection(aws_access_key_id, aws_secret_access_key)
-        except Exception as ex:
-            raise NotConfigured(str(ex))
```

Comment:
So, in case of error it will won't disable the extension, instead it fails every time the handler is used. am I right?